packages/astro/src/core/render/dev/resolve.ts

```diff
@@ -1,17 +1,17 @@
 import type { ModuleLoader } from '../../module-loader/index';
 import { resolveIdToUrl } from '../../util.js';
 
-export function createResolve(loader: ModuleLoader) {
+export function createResolve(loader: ModuleLoader, root: URL) {
 	// Resolves specifiers in the inline hydrated scripts, such as:
 	// - @astrojs/preact/client.js
 	// - @/components/Foo.vue
 	// - /Users/macos/project/src/Foo.vue
 	// - C:/Windows/project/src/Foo.vue (normalized slash)
 	return async function (s: string) {
-		const url = await resolveIdToUrl(loader, s);
+		const url = await resolveIdToUrl(loader, s, root);
 		// Vite does not resolve .jsx -> .tsx when coming from hydration script import,
 		// clip it so Vite is able to resolve implicitly.
-		if (url.startsWith('/@fs') && url.endsWith('.jsx')) {
+		if (url.startsWith('/') && url.endsWith('.jsx')) {
```

Comment:
Changed to a simple slash check, but I think this shouldn't cause issues in practice. This is needed because we need to strip `.jsx` for root-relative paths too, that doesn't start with `/@fs`